tests/Unit/SDK/Common/Util/DanglingSpansShutdownHandlerTest.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Tests\Unit\SDK\Common\Util;
+
+use OpenTelemetry\SDK\Common\Util\DanglingSpansShutdownHandler;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\API\Trace\NoopTracer;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \OpenTelemetry\SDK\Common\Util\DanglingSpansShutdownHandler
+ */
+class DanglingSpansShutdownHelperTest extends TestCase
+{
+    public function test_cleanup(): void
+    {
+        $tracer = NoopTracer::getInstance();
+        $span1 = $tracer->spanBuilder("span1")->startSpan();
+        $scope1 = $span1->activate();
+        $this->assertNotEquals(Context::getCurrent(), Context::getRoot());
+        // Intentionally no detach call
+
+        // Will trigger_error without this
+        DanglingSpansShutdownHandler::shutdown();
```

Comment:
I think another assertion here that context now _is_ root?

Reply:
will do... I wanted to add an assertion for the span being ended, but the NoopTracers spans don't really do anything.  I can try to mock out the various things to include the span end assertion as well.

Reply:
I get really lost when I try to figure out how to create a span that sets the context.  I added your assert but haven't figured out how to simulate the span so I can detect the end call.  I can't even really tell how this code is creating a new context given the nonRecordingSpans involved

Reply:
I'd look at mockery here, eg mocking a tracer, then having its spanBuilder method return a partial mock of a span: https://docs.mockery.io/en/latest/reference/partial_mocks.html - you should then be able to assert on the span's end() method.